### PR TITLE
Cleanup of pom names and relative paths to avoid warnings

### PR DESCRIPTION
**Pull request by spericas**
Cleanup of pom names, and changes to relative paths to avoid warnings when building modules. Some minor updates to Javadocs to avoid warnings as well.

Requesting fast track.

### CODE REVIEW

**Review by mkarg — DISMISSED** [review]

**Review by andymc12 — APPROVED** [review]

File: jaxrs-api/src/main/java/jakarta/ws/rs/core/EntityPart.java [hunk 1]
```diff
@@ -211,7 +213,7 @@ static Builder withFileName(String partAndFileName) {
      * 
      * @since 3.1
      */
-    public interface Builder {
```
Comment (mkarg) [inline jaxrs-api/src/main/java/jakarta/ws/rs/core/EntityPart.java:214]: Why removing `public`?

**Review by andymc12 — DISMISSED** [review]
LGTM - thanks!

File: jaxrs-api/src/main/java/jakarta/ws/rs/core/EntityPart.java (hunk 1 above)
Comment (spericas) [reply jaxrs-api/src/main/java/jakarta/ws/rs/core/EntityPart.java:214]: It's redundant for inner interfaces, as it is for methods etc. in all interfaces.

**Review by chkal — APPROVED** [review]
Thanks! But it looks like there are merge conflicts in one of the pom.xml files.

**Review by andymc12 — APPROVED** [review]

**Review by chkal — APPROVED** [review]

**Pull request merged**